### PR TITLE
security: add size limit to file uploads and lobbies

### PR DESCRIPTION
**Pull request by mredstone6**
# Feature:
- add limit to file upload
- add limit to lobby

### CODE REVIEW

**Comment by Tran-Antoine** [comment]
I'm not a huge fan of the DiscordContext being passed as context parameter

**Review by Tran-Antoine — CHANGES_REQUESTED** [review]
In every command in which you use the CounterLimiter, I think it would be much better to include its usage in the  `StopConditions.shouldStop` system

**Comment by Tran-Antoine** [comment]
This also contains a memory leak fix.

File: discord-impl/src/main/java/net/starype/quiz/discordimpl/command/StartGameCommand.java [hunk 1]
```diff
@@ -26,30 +30,36 @@ public void execute(CommandContext context) {
...
                 () -> lobbyList.findByPlayer(playerId).isEmpty(),
                 nickName + ", you are not in any lobby");
+
         conditions.put(
                 () -> lobbyList.findByAuthor(playerId).isEmpty(),
                 nickName + ", only the owner of the lobby can start the game");
 
+        conditions.put(
+                () -> !gameLimiter.acquireInstance(uniqueId),
+                "Error: Cannot create a new game as the maximum number of game as been reached");
```
Comment (mredstone6) [inline discord-impl/src/main/java/net/starype/quiz/discordimpl/command/StartGameCommand.java:61]:
```suggestion
                "Error: Cannot create a new game as the maximum number of game has been reached");

```

File: discord-impl/src/main/java/net/starype/quiz/discordimpl/command/CreateLobbyCommand.java [hunk 2]
```diff
@@ -48,6 +52,11 @@ public void execute(CommandContext context) {
         conditions.put(
                 () -> gameList.isPlaying(authorId),
                 nickName + ", you are already playing a game");
+
+        conditions.put(
+               () -> !lobbyLimiter.acquireInstance(authorId.hashCode()),
+               "max lobby limit has been reached");
```
Comment (mredstone6) [inline discord-impl/src/main/java/net/starype/quiz/discordimpl/command/CreateLobbyCommand.java:58]:
```suggestion
               "Error: Cannot create new lobby as the maximum number of lobbies has been reached");

```

File: discord-impl/src/main/java/net/starype/quiz/discordimpl/util/InputUtils.java [hunk 3]
```diff
@@ -16,23 +18,32 @@
...
+
     public static Collection<? extends EntryUpdater> loadEntryUpdaters(String urlName, TextChannel channel) {
         Set<EntryUpdater> updaters = new HashSet<>();
 
         try {
             URL url = new URL(urlName);
-            InputStream fileStream = new BufferedInputStream(url.openStream(), 1024);
-            ZipInputStream zipStream = new ZipInputStream(fileStream);
+            if(!downloadingLimiter.acquireInstance(urlName.hashCode())) {
+                channel.sendMessage("Error: The limit of downloading zip as been reached").queue();
```
Comment (firestar24) [inline discord-impl/src/main/java/net/starype/quiz/discordimpl/util/InputUtils.java:31]:
```suggestion
                channel.sendMessage("Error: The limit of downloading zip has been reached").queue();

```

**Review by Tran-Antoine — APPROVED** [review]

**Pull request merged**